source/features/conversation-timeline-filter.tsx

```diff
@@ -0,0 +1,225 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import elementReady from 'element-ready';
+import {observe} from 'selector-observer';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+enum FilterSettings {
+	ShowAll = 1,
+	ShowOnlyComments = 2,
+	ShowOnlyUnresolvedComments = 3,
+	ShowOnlyUnresolvedReviews = 4
+}
+
+let currentSettings: FilterSettings = FilterSettings.ShowAll;
+
+const showFilterName = 'rgh-show-filter';
+const filterFormId = 'show-filter-form';
+
+// Every element on the timeline that is recognizable by this feature will be marked with tis class.
+const timelineFiltersSelectorId = 'timeline-filters';
+const detailsSelector = `#${timelineFiltersSelectorId} details`;
+const notiticationsSelector = '.discussion-sidebar-item.sidebar-notifications';
+const timelineItemSelector = '.js-timeline-item';
+
+function regenerateFilterSummary(): void {
+	const timelineFilter = select(`#${timelineFiltersSelectorId}`)!;
+
+	let text = '';
+	switch (currentSettings) {
+		case FilterSettings.ShowAll:
+			text = 'Show All';
+			break;
+		case FilterSettings.ShowOnlyComments:
+			text = 'Show only comments';
+			break;
+		case FilterSettings.ShowOnlyUnresolvedReviews:
+			text = 'Show only unresolved reviews';
+			break;
+		case FilterSettings.ShowOnlyUnresolvedComments:
+			text = 'Show unresolved comments';
+			break;
+		default:
+			text = '';
+			break;
+	}
+
+	const newSummary = (
+		<p className="reason text-small text-gray">
+			{text}
+		</p>
+	);
+
+	select('p.reason', timelineFilter)!.replaceWith(newSummary);
+}
```

Comment:
There must be a better way to handle this, switch-less and enum-less. For example:


```js
const copy = {
	ShowAll: 'Show all',
	ShowOnlyUnresolvedComments: 'Show unresolved comments',
};

function regenerateFilterSummary(value) {
  	select('a single selector here .reason')!.textContent = copy[value];
}
```

Reply:
Done